core/API/Request.php

```diff
@@ -313,8 +313,26 @@ public static function setIsRootRequestApiRequest($isRootRequestApiRequest)
      */
     public static function isRootRequestApiRequest()
     {
-        $isApi = Cache::getTransientCache()->fetch('API.setIsRootRequestApiRequest');
-        return !empty($isApi);
+        $apiMethod = Cache::getTransientCache()->fetch('API.setIsRootRequestApiRequest');
+        return !empty($apiMethod);
+    }
+
+    /**
+     * Checks if the API method of the root API request is == to $methodToCheck.
+     *
+     * @param string|null $methodToCheck The API method to check, eg, VisitsSummary.get. If null, gets the
+     *                                   current value of the 'method' query parameter.
+     * @return bool
+     * @throws Exception
+     */
+    public static function isRootApiRequestHandlingMethod($methodToCheck = null)
+    {
+        if (empty($methodToCheck)) {
+            $methodToCheck = Common::getRequestVar('method', '', 'string');
```

Comment:
I'm not quite sure if this works / makes too much sense? `$_GET/$_POST['method']` will be overwritten during API calls within the code etc. And it seems to be only called once but without a parameter? And what is the difference to `isRootRequestApiRequest()` when called without a parameter or so? I've tried to understand what is happening here but couldn't.